drivers/adc/adc_stm32.c

```diff
@@ -296,6 +272,8 @@ struct adc_stm32_cfg {
 	int8_t temp_channel;
 	int8_t vref_channel;
 	int8_t vbat_channel;
+	int8_t res_table_size;
+	const uint32_t *res_table;
```

Comment:
This can be packed a little more efficiently, without the extra pointer, by using a flexible array member, which is part of the C99 standard, via:

```c
struct adc_stm32_cfg {
	ADC_TypeDef *base;
	…
	int8_t res_table_size;
	const uint32_t res_table[];
};

static const struct adc_stm32_cfg adc_stm32_cfg_##index = {		\
	…
	.res_table_size = DT_INST_PROP_LEN(index, resolutions),		\
	.res_table = DT_INST_PROP(index, resolutions),			\
};									\

Reply:
Done